### ISSUE TITLE
Unstable __eq__ implementation on pages

### ISSUE THREAD

**Issue by jkuball** [pikepdf/pikepdf#374]
Hey!

I recently had the problem where I used `itertools.zip_longest()` to mix some pages -- and to check whether it used the fillvalue, I tried to check it with the `in` operator. Then it crashed because it always assumes both operands have the attribute `obj` set.

Here is a small example to reproduce:

```pycon
>>> import pikepdf
>>> pikepdf.open('/some/pdf')
>>> pdf.pages
<pikepdf._qpdf.PageList len=14>
>>> None in pdf.pages
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/spam/Library/Caches/pypoetry/virtualenvs/eggs-JBqLZSwl-py3.10/lib/python3.10/site-packages/pikepdf/_methods.py", line 1134, in __eq__
    return self.obj == other.obj
AttributeError: 'NoneType' object has no attribute 'obj'
```

I don't think an equals operator should raise an Exception ever, so I'd love to see this fixed!

(Note: I fixed it with using `any(page is None for page in pdf.pages)`)

I am using Pikepdf 5.4.1, but it looks like it is in the library for quite some time now.

Have a nice day!

**Comment by mara004** [comment]
[Here](https://github.com/pikepdf/pikepdf/blob/2a4bd6ae0f1c510e8262e4d67cf3ae0fd6988685/src/pikepdf/_methods.py#L1133)'s the definition. You could submit a Pull Request if you like.
I guess one could either compare types first, or check if `other` provides `.obj` using `hasattr()`.

**Comment by jkuball** [comment]
Yes, found that, just did not have the time to fix it myself. Usually I would prefer isinstance(), but I am not sure how this would work with the @augments stuff since I am not familiar with it. Maybe, in this case, an early `if not hasattr(other, 'obj'): return False` is the best way.
...

**Comment by jbarlow83** [comment]
Thanks for this - fixed in v5.4.2
